src/server/electron/electron.ts

```diff
@@ -100,19 +100,13 @@ export class ElectronApplication extends SdkObject {
   }
 
   async close() {
-    const closed = this._waitForEvent(ElectronApplication.Events.Close);
+    const progressController = new ProgressController(internalCallMetadata(), this);
+    const closed = progressController.run(progress => helper.waitForEvent(progress, this, ElectronApplication.Events.Close).promise, this._timeoutSettings.timeout({}));
     await this._nodeElectronHandle!.evaluate(({ app }) => app.quit());
     this._nodeConnection.close();
     await closed;
   }
 
-  private async _waitForEvent(event: string, predicate?: Function): Promise<any> {
-    const progressController = new ProgressController(internalCallMetadata(), this);
-    if (event !== ElectronApplication.Events.Close)
-      this._browserContext._closePromise.then(error => progressController.abort(error));
```

Comment:
Why do we no longer need it?

Reply:
We only called this method with 'close' event.